src/app/(app)/(home)/search-filters/categories.tsx

```
@@ -0,0 +1,115 @@
+export const Categories = ({ data }: Props) => {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const measurerRef = useRef<HTMLDivElement>(null);
+  const viewAllRef = useRef<HTMLDivElement>(null);
+
+  const [visibleCount, setVisibleCount] = useState(data.length);
+  const [isAnyHovered, setIsAnyHovered] = useState(false);
+  const [isSidebarOpen, setIsSidebarOpen] = useState(false);
+
+  const activeCategory = 'all';
+  const activeCategoryIndex = data.findIndex((cat) => {
+    cat.slug === activeCategory;
```

ESLint

[error] Expected an assignment or function call and instead saw an expression.

(@typescript-eslint/no-unused-expressions)